### PR TITLE
Feature/routes

### PR DESCRIPTION
**Pull request by alexfritsch10**
This PR includes:

- Complete setup of the mongoDB by adding environment data
- Document API Endpoints in the README.md
- Fix functionality of the DELETE /wiki/{articleIID} and the POST /wiki endpoint

### CODE REVIEW

File: MongoDBAPI.py [hunk 1]
```diff
@@ -1,70 +1,90 @@
...
+        collection = "articles"
+        db = self.client[database]
+        # creates collection in wiki database
+        if collection not in db.list_collection_names():
+            db.create_collection(collection)
+        self.collection = db[collection]
 
     def read(self):
-        log.info('Reading All Data')
+        log.info('Reading all Articles')
```
Comment (jgraeger) [inline MongoDBAPI.py:26]:
```suggestion
        log.info('Reading all articles')
```

File: MongoDBAPI.py [hunk 2]
```diff
@@ -1,70 +1,90 @@
...
         documents = self.collection.find()
         output = [{item: data[item] for item in data if item != '_id'} for data in documents]
         return output
 
     def write(self, data):
-        log.info('Writing Data')
-        new_document = data['Document']
-        response = self.collection.insert_one(new_document)
+        log.info('Writing one Article')
```
Comment (jgraeger) [inline MongoDBAPI.py:32]:
```suggestion
        log.info('Writing one article')
```

File: MongoDBAPI.py [hunk 3]
```diff
@@ -1,70 +1,90 @@
...
+        response = self.collection.insert_one({"title": data["title"], "content": data["content"]})
         output = {'Status': 'Successfully Inserted',
                   'Document_ID': str(response.inserted_id)}
         return output
 
+    def delete(self, articleID):
+        log.info('Deleting one Article')
```
Comment (jgraeger) [inline MongoDBAPI.py:39]:
```suggestion
        log.info('Deleting one article')
```

File: MongoDBAPI.py [hunk 4]
```diff
@@ -1,70 +1,90 @@
...
+        response = self.collection.delete_one({"_id": ObjectId(articleID)})
+        output = {'DeletedCount': str(response.deleted_count)}
+        if response.deleted_count == 1:
+            output["Status"] = 'Successfully Deleted'
+        else:
+            output["Status"] = 'Could not delete object'
```
Comment (jgraeger) [inline MongoDBAPI.py:45]:
```suggestion
        output["success"] = response.deleted_count == 1
```

File: MongoDBAPI.py [hunk 5]
```diff
@@ -1,70 +1,90 @@
...
     return Response(response=json.dumps(response),
-                    status=200,
+                    status=HTTPStatus.OK,
+                    mimetype='application/json')
+
+
+@app.route('/wiki/<articleID>', methods=['DELETE'])
+def wiki_deleteArticle(articleID):
+    obj1 = MongoAPI()
+    response = obj1.delete(articleID)
```
Comment (jgraeger) [inline MongoDBAPI.py:83]:
```suggestion
    db = MongoAPI()
    response = db.delete(articleID)
```

File: MongoDBAPI.py [hunk 6]
```diff
@@ -1,70 +1,90 @@
...
-        return Response(response=json.dumps({"Error": "Please provide connection information"}),
-                        status=400,
+    # expects "title" and "content" field in body
+    if 'title' not in data or 'content' not in data:
+        return Response(response=json.dumps({"Error": "Please provide article information (title and content)"}),
+                        status=HTTPStatus.BAD_REQUEST,
                         mimetype='application/json')
-    obj1 = MongoAPI(data)
+    obj1 = MongoAPI()
     response = obj1.write(data)
```
Comment (jgraeger) [inline MongoDBAPI.py:74]:
```suggestion
    db = MongoAPI()
    response = db.write(data)
```

File: MongoDBAPI.py [hunk 7]
```diff
@@ -1,70 +1,90 @@
...
     return Response(response=json.dumps(response),
-                    status=200,
+                    status=HTTPStatus.OK,
```
Comment (jgraeger) [inline MongoDBAPI.py:76]:
```suggestion
                    status=HTTPStatus.CREATED,
```

For creating a new ressource, return HTTP Status Code `201 - Created`.

File: MongoDBAPI.py [hunk 8]
```diff
@@ -1,70 +1,90 @@
...
-    data = request.json
-    if data is None or data == {}:
-        return Response(response=json.dumps({"Error": "Please provide connection information"}),
-                        status=400,
-                        mimetype='application/json')
-    obj1 = MongoAPI(data)
+@app.route('/wiki', methods=['GET'])
+def wiki_readAllArticles():
+    obj1 = MongoAPI()
     response = obj1.read()
```
Comment (jgraeger) [inline MongoDBAPI.py:59]:
```suggestion
    db = MongoAPI()
    response = db.read()
```

**Review by jgraeger — APPROVED** [review]
LG. I added some small suggestions :)

For some reason, GitHub seems unable to perform an automatic merge, so please resolve conflicts manually during the merge process. Ty!

**Comment by akkbng** [comment]
Hi @alexfritsch10, could you please check the suggestions from @jgraeger and resolve the conflicts? After merging this branch we can check which issues/tickets are still to be implemented, thanks!

**Pull request merged**